pkg/costmodel/cluster.go

```diff
@@ -297,6 +305,34 @@ func ClusterDisks(client prometheus.Client, provider cloud.Provider, start, end
 		diskMap[key].Minutes = mins
 	}
 
+	//Iterating through Persistent Volume given by custom metrics kubecost_pv_info and assign the storage class if known and __unknown__ if not populated.
+	for _, result := range resPVStorageClass {
+		cluster, err := result.GetString(env.GetPromClusterLabel())
+		if err != nil {
+			cluster = env.GetClusterID()
+		}
+
+		name, _ := result.GetString("persistentvolume")
+
+		key := DiskIdentifier{cluster, name}
+		if _, ok := diskMap[key]; !ok {
+			log.DedupedWarningf(5, "ClusterDisks: Storage class information for unidentified disk or disk deleted from analysis")
+			continue
+		}
```

Comment:
If this is worth a log, let's also include the cluster/name combo in the log message!

Reply:
In light of our log discussion last week, what do you think about storing this warning info and then logging a summary outside of the loop? I definitely won't block this PR on this, just an idea if you feel inspired :slightly_smiling_face: